views/find_missing.py

```diff
@@ -92,9 +115,137 @@ def render_search_button(self):
         btn_layout.addWidget(self.search_button)
         self.addLayout(btn_layout)
 
-    def search(self):
-        # Search through fetched cards here
-        deck_id = self.deck_selection_enabled and self.deck_chooser.selectedId()
+    def render_search_preview(self):
+        search_preview_layout = QHBoxLayout()
+        search_text = QLabel()
+        search_text.setText("Search:")
+        search_preview_layout.addWidget(search_text)
+        
+        self.search_preview = QTextEdit()
+        self.search_preview.setReadOnly(True)
+        self.search_preview.setFixedHeight(40)
+        search_preview_layout.addWidget(self.search_preview)
+
+        # self.copy_preview_button = QPushButton("Copy")
+        # self.copy_preview_button.clicked.connect(self.copy_preview)
+        # search_preview_layout.addWidget(self.copy_preview_button)
+
+        self.addLayout(search_preview_layout)
+
+    def render_results_area(self):
+        self.results_area = QListWidget()
+        self.addWidget(self.results_area)
+
+
+
+########################
+#######  Non-UI  #######
+########################
+
+
+
+    # def copy_preview(self):
+    #     data = self.get_final_search("[Word]")
+    #     command = 'echo ' + data.strip() + '| clip'
+    #     os.system(command)
+    
+    def update_init_search(self):
+        # print("Update Init Search")
         deck_name = self.deck_selection_enabled and self.deck_chooser.deckName()
-        model_fields = self.model_field_selection_enabled and self.model_field_chooser.selected_items
+        model_fields = self.model_field_selection_enabled and self.model_field_selected_items
+
+        if self.model_field_selection_enabled:
+            noteTypesSelected = [mod['name'] for mod in model_fields] if model_fields else ""
+            fieldsSelected = list({fields['name'] for mod in model_fields for fields in mod['fields']} if model_fields else "")
+            
+            print(noteTypesSelected, fieldsSelected)
 
+        self.initQuery = ""
+        self.initQuery += self.search_formatter(True, "deck", deck_name) if deck_name else ""
+        self.initQuery += self.search_multiple_terms_formatter("note", noteTypesSelected) if self.model_field_selection_enabled else ""
+        self.initQuery += self.search_multiple_fields_formatter(fieldsSelected, self.REPLACEMENT_STRING) if self.model_field_selection_enabled else ""
```

Comment:
Not sure if a more general search is going in the second pass, but I have trouble with the search if I don't filter on model/field because the replacement string is missing. Perhaps:

```
if not self.model_field_selection_enabled:
   self.initQuery += ' ' + self.REPLACEMENT_STRING
```

Reply:
Oh, you are right, I totally missed this in the merge! Fixing now :+1: